File: packages/android/src/main/java/com/kozmos/components/BottomSheet/BottomSheet.figma.kt
package com.kozmos.components.bottomsheet

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType
import com.figma.code.connect.FigmaVariant

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=624-4362")
@FigmaVariant("Content", "Basic")
class KozmosBottomSheetBasicConnect {
    @FigmaProperty(FigmaType.Text, "Title Text")
    val title: String = "Sheet title"

    @FigmaProperty(FigmaType.Text, "Description Text")
    val description: String = "Sheet supporting content."

    @Composable
    fun ComponentExample() {
        KozmosBottomSheet(onDismissRequest = {}) {
            Text(title)
            Text(description)
        }
    }
}

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=624-4362")
@FigmaVariant("Content", "Form")
class KozmosBottomSheetFormConnect {
    @FigmaProperty(FigmaType.Text, "Title Text")
    val title: String = "Route details"

    @FigmaProperty(FigmaType.Text, "Description Text")
    val description: String = "Review nearby stops and choose what to do next."

    @Composable
    fun ComponentExample() {
        KozmosBottomSheet(onDismissRequest = {}) {
            Text(title)
            Text(description)
            Text("Search places")
            Text("Optional context")
        }
    }
}

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=624-4362")
@FigmaVariant("Content", "Footer")
class KozmosBottomSheetFooterConnect {
    @FigmaProperty(FigmaType.Text, "Title Text")
    val title: String = "Route details"

    @FigmaProperty(FigmaType.Text, "Description Text")
    val description: String = "Review nearby stops and choose what to do next."

    @Composable
    fun ComponentExample() {
        KozmosBottomSheet(onDismissRequest = {}) {
            Text(title)
            Text(description)
            Text("Cancel")
            Text("Apply")
        }
    }
}

File: packages/ios/Sources/Components/BottomSheet/BottomSheet.figma.swift
import SwiftUI
import Figma

struct KozmosBottomSheetBasicConnect: FigmaConnect {
    let component = KozmosBottomSheet<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=624-4362"
    var variant = ["Content": "Basic"]

    @FigmaString("Title Text")
    var title: String = "Sheet title"

    @FigmaString("Description Text")
    var description: String = "Sheet supporting content."

    var body: some View {
        KozmosBottomSheet(isPresented: .constant(true)) {
            AnyView(
                VStack(alignment: .leading, spacing: KozmosDimensions.primitivesLayoutSpacing200) {
                    Text(self.title)
                        .font(KozmosTypography.headline)
                    Text(self.description)
                        .foregroundColor(KozmosColors.primitivesColorsForeground500)
                }
                .padding(KozmosDimensions.primitivesLayoutSpacing400)
            )
        }
    }
}

struct KozmosBottomSheetFormConnect: FigmaConnect {
    let component = KozmosBottomSheet<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=624-4362"
    var variant = ["Content": "Form"]

    @FigmaString("Title Text")
    var title: String = "Route details"

    @FigmaString("Description Text")
    var description: String = "Review nearby stops and choose what to do next."

    var body: some View {
        KozmosBottomSheet(isPresented: .constant(true)) {
            AnyView(
                VStack(alignment: .leading, spacing: KozmosDimensions.primitivesLayoutSpacing300) {
                    Text(self.title)
                        .font(KozmosTypography.headline)
                    Text(self.description)
                        .foregroundColor(KozmosColors.primitivesColorsForeground500)
                    TextField("Search places", text: .constant(""))
                    TextField("Optional context", text: .constant(""))
                }
                .padding(KozmosDimensions.primitivesLayoutSpacing400)
            )
        }
    }
}

struct KozmosBottomSheetFooterConnect: FigmaConnect {
    let component = KozmosBottomSheet<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=624-4362"
    var variant = ["Content": "Footer"]

    @FigmaString("Title Text")
    var title: String = "Route details"

    @FigmaString("Description Text")
    var description: String = "Review nearby stops and choose what to do next."

    var body: some View {
        KozmosBottomSheet(isPresented: .constant(true)) {
            AnyView(
                VStack(alignment: .leading, spacing: KozmosDimensions.primitivesLayoutSpacing300) {
                    Text(self.title)
                        .font(KozmosTypography.headline)
                    Text(self.description)
                        .foregroundColor(KozmosColors.primitivesColorsForeground500)
                    HStack {
                        Button("Cancel") {}
                        Button("Apply") {}
                    }
                }
                .padding(KozmosDimensions.primitivesLayoutSpacing400)
            )
        }
    }
}
